src/rail_cmd.cpp

```diff
@@ -129,15 +129,14 @@ static bool CompareRailTypes(const RailType &first, const RailType &second)
  */
 void InitRailTypes()
 {
-	for (RailType rt = RAILTYPE_BEGIN; rt != RAILTYPE_END; rt++) {
-		RailTypeInfo *rti = &_railtypes[rt];
-		ResolveRailTypeGUISprites(rti);
-		if (rti->flags.Test(RailTypeFlag::Hidden)) _railtypes_hidden_mask.Set(rt);
-	}
-
 	_sorted_railtypes.clear();
-	for (RailType rt = RAILTYPE_BEGIN; rt != RAILTYPE_END; rt++) {
-		if (_railtypes[rt].label == 0) continue;
+	for (auto it = std::begin(_railtypes); it != std::end(_railtypes); ++it) {
+		RailType rt = static_cast<RailType>(it - std::begin(_railtypes));
```

Comment:
I'm not convinced this is an improvement

Reply:
Fair enough. A different approach...